### ISSUE TITLE
CodeLens doesn't appear after renaming a file

### ISSUE THREAD

**Issue by angelozerr** [redhat-developer/lsp4ij#12]
After renaming Java file which have codelens, sometimes the codelens are not refreshed and there are this error:

```
sept. 12, 2023 2:20:21 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: java.lang.reflect.InvocationTargetException
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
	at org.eclipse.lsp4mp.ls.commons.ParentProcessWatcher.lambda$apply$0(ParentProcessWatcher.java:148)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	... 12 more
Caused by: java.lang.NullPointerException: Cannot invoke "org.eclipse.lsp4mp.ls.java.JavaTextDocuments$JavaTextDocument.executeIfInMicroProfileProject(java.util.function.BiFunction, Object)" because "document" is null
	at org.eclipse.lsp4mp.ls.java.JavaFileTextDocumentService.codeLens(JavaFileTextDocumentService.java:203)
	at org.eclipse.lsp4mp.ls.MicroProfileTextDocumentService.codeLens(MicroProfileTextDocumentService.java:240)
	... 17 more

Caused by: java.lang.reflect.InvocationTargetException

Caused by: java.lang.NullPointerException: Cannot invoke "org.eclipse.lsp4mp.ls.java.JavaTextDocuments$JavaTextDocument.executeIfInMicroProfileProject(java.util.function.BiFunction, Object)" because "document" is null


sept. 12, 2023 2:27:10 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: java.lang.reflect.InvocationTargetException
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
	at org.eclipse.lsp4mp.ls.commons.ParentProcessWatcher.lambda$apply$0(ParentProcessWatcher.java:148)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	... 12 more
Caused by: java.lang.NullPointerException: Cannot invoke "org.eclipse.lsp4mp.ls.java.JavaTextDocuments$JavaTextDocument.executeIfInMicroProfileProject(java.util.function.BiFunction, Object)" because "document" is null
	at org.eclipse.lsp4mp.ls.java.JavaFileTextDocumentService.codeLens(JavaFileTextDocumentService.java:203)
	at org.eclipse.lsp4mp.ls.MicroProfileTextDocumentService.codeLens(MicroProfileTextDocumentService.java:240)
	... 17 more

Caused by: java.lang.reflect.InvocationTargetException

Caused by: java.lang.NullPointerException: Cannot invoke "org.eclipse.lsp4mp.ls.java.JavaTextDocuments$JavaTextDocument.executeIfInMicroProfileProject(java.util.function.BiFunction, Object)" because "document" is null


2023-09-12 14:33:23,083 [13940383]   WARN - org.eclipse.lsp4j.jsonrpc.RemoteEndpoint - Unmatched cancel notification for request id 58

```

This error means that document doesn't exists in LS side, but after some checkn when a Java file is renamed, there are:

 * a didOpen with the new renamed file name
 * a codelens triggered with the new  renamed file name

But when I debug the MicroProfile LS, codelens service is called after a didOpen?

**Comment by fbricon** [comment]
would #173 fix this?

**Comment by angelozerr** [comment]
Indeed it should fix it.

We should fix too MP LS to throw an error or return an empty codelens when didOpen is not called.

**Comment by fbricon** [comment]
Fixed with #173
